### DISCUSSION TITLE
Phoebus Resync / Planning Meeting - Fri. 13, Dec., 2024

### DISCUSSION THREAD

**Discussion by carlnotsagan** [lanl/phoebus#221] [Ideas]
### Current development
- Working to update with new parthenon - a lot of refactoring/ particle features
- Working branch - no PR yet 

### Current issues/challenges
- Major issues solved with Kokkos 4.5 Cray machines - parthenon working with updated Kokkos 
- Additional challenges with currently running on LANL machines
- No View of Views unless u need to - wrong steps given by Kokkos docs. 
- Possible issues in tasking
- Last big change left at API level - make sure particles work with AMR only with SMR now 
- Riemann solver needs rewritten - can't convert it to sparse packs 
- Testing infrastructure limited by Github
- Good to have access to other hardware - external server - possible at one of the IHE? Not arizona for now..
- 3D CCSN Lightbulb challenges - poor performance, inconsistant results?
- MC particles infrastructure needs work
- Many other physics pieces not being developed
- Docs incomplete - need more details covering state of the code. 
- Long term challenge: create an explicit feature policy. Example: someone implements a feature that needs maintaining in the future and interfaces with many pieces of the code. 

### Possible action items
- Ben P. and Brandon planing a meeting to work on phoebus up-to-date with current parthenon using lessons learned from kharma.  
- Detailed developers guide in the Docs but most via Github Issues possibly started from Github Discussions. 
- Developers Guide could include "lessons learned" or design philosophy/science-driven code choices and their discussions (Singularity-EOS a good starting point for this again). 
- More physics tests for current capabilities - how do we know the code works???
- Detailed Tutorial for adding/contributing to the phoebus docs either directly or from a Discussion or merged PR/Issue.
- Detail new options for development / collaboration on the README
- CI set for parthenon for MHD etc, could use Philips machine with HPC lab or migrate to HPC LANL namespace
- Miah opening phoebus slack workspace to public - might need to do on the web browser. Fine for now given small project size. 
- Send a when2meet end of the year/first week of Jan. to schedule recurring meeting - acknowledging that core developers will meet less often. Need: clear notes at each meeting to keep those that cannot meet in sync. Scribe?

### General Comments
- Parthenon refactoring not a persistent problem - growing pain - approaching state of maturity. Good change. "We know what we are doing now."
- Need  for additional consistent development for Phoebus. 
- Anyone with one or a few commits since the first instrument paper should/could be a co-author on possilbe second instrument paper. 
- Kharma documentation been helpful for many things - strong support for improving the docs!
- Brief overview of Phoebus proposals submitted - AAG, CSSI - would support grad students to help with docs, infrastructure, and development, and/or science topics in CCSNe. 
- Focus needs to be placed on development challenges.
- Less telephone and additional meetings, steps to stay synched. 
- Strongly discourage Github Forks. 
- All bug reports / features should continue to stay in github Issues. 
- Goal/Science-driven development is critical. 

### Meeting outcomes
- Started Github Discussions for: open discussions on questions, development, challenges then should be moved to a Github Issues or Docs. Goal: reduce barriers to discussing challenges openly

**Comment by Yurlungur** [comment]
Thanks for the summary @carlnotsagan !
